src/dc_tools.rs

```diff
@@ -330,13 +330,13 @@ fn encode_66bits_as_base64(v1: u32, v2: u32, fill: u32) -> String {
     let mut wrapped_writer = Vec::new();
     {
         let mut enc = base64::write::EncoderWriter::new(&mut wrapped_writer, base64::URL_SAFE);
-        enc.write_u32::<BigEndian>(v1).unwrap();
-        enc.write_u32::<BigEndian>(v2).unwrap();
-        enc.write_u8(((fill & 0x3) as u8) << 6).unwrap();
-        enc.finish().unwrap();
+        enc.write_u32::<BigEndian>(v1).unwrap_or_default();
+        enc.write_u32::<BigEndian>(v2).unwrap_or_default();
+        enc.write_u8(((fill & 0x3) as u8) << 6).unwrap_or_default();
+        enc.finish().unwrap_or_default();
     }
     assert_eq!(wrapped_writer.pop(), Some(b'A')); // Remove last "A"
-    String::from_utf8(wrapped_writer).unwrap()
+    String::from_utf8(wrapped_writer).unwrap_or_default()
```

Comment:
IIUC we just wrote a bunch of base64-encoded stuff into a vector.  This is guaranteed to be utf-8 and therefore an unwrap was correct here.